### PR TITLE
[Server+Client] Claim hexalots

### PR DESCRIPTION
**Pull request by mitchelljustin**
Seems like client is broken on master, so can't actually test my claim-lot logic yet.

Let's talk tomorrow (Mar 19)

### CODE REVIEW

File: client/src/view/gotchi-view.tsx [hunk 1]
```diff
@@ -116,7 +116,7 @@ export class GotchiView extends React.Component<IGotchiViewProps, IGotchiViewSta
                                 />
                             </R3.Object3D>
                         )}
-                        {!journey ? null : (
+                        {!journey ? undefined : (
```
Comment (geralddejong) [inline client/src/view/gotchi-view.tsx:119]: ah yes.  cool to be more consistent.

File: server/src/router.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        .route("/journey")
+        .get(async (req, res) => {
+            const rotation = await store.getRotation(res.locals.hexalotId)
+            res.json(rotation)
+        })
+        .post(
+            [
+                body("rotation").isNumeric().toFloat(),
+            ],
+            async (req: Request, res: Response) => {
```
Comment (geralddejong) [inline server/src/router.ts:150]: hmmm.. gonna have to look this up.  stuff i haven't used.

File: server/src/store.ts [hunk 3]
```diff
@@ -36,44 +34,53 @@ export class LevelDBFlashStore implements IKeyValueStore {
...
-    private async get(key: string): Promise<any | null> {
-        return this.db.get(`/island/${this.islandName}/${key}`)
+    public async setGenome(id: HexalotID, genomeData: string): Promise<void> {
+        return this.db.set(`/hexalot/${id}/genomeData`, genomeData)
     }
+
+    public async getRotation(id: HexalotID): Promise<number> {
+        let rotation = await this.db.get(`/hexalot/${id}/rotation`)
+        if (rotation === undefined) {
+            rotation = 0 // TODO: default rotation?
```
Comment (geralddejong) [inline server/src/store.ts]: yes. 0 to 5. hexagon direction

**Comment by geralddejong** [comment]
Was the broken client a Lint issue?  I committed a fix to index.tsx to master.

File: client/src/view/gotchi-view.tsx (hunk 1 above)
Comment (geralddejong) [reply client/src/view/gotchi-view.tsx:119]: found a few more of these, fixed 'em in your branch

File: server/src/router.ts (hunk 2 above)
Comment (mitchelljustin) [reply server/src/router.ts:150]: You mean middlewares & whatnot? Pretty cool

File: server/src/store.ts (hunk 3 above)
Comment (mitchelljustin) [reply server/src/store.ts]: great

**Comment by mitchelljustin** [comment]
@geralddejong You might be right on the lint issue. I'll merge master

File: client/src/view/gotchi-view.tsx (hunk 1 above)
Comment (mitchelljustin) [reply client/src/view/gotchi-view.tsx:119]: :) thx

**Comment by geralddejong** [comment]
I already merged master in here.

File: server/src/router.ts (hunk 2 above)
Comment (geralddejong) [reply server/src/router.ts:150]: async and await

Comment (mitchelljustin) [reply server/src/router.ts:150]: it's so much nicer than callback or Promise hell

**Comment by mitchelljustin** [comment]
@geralddejong Throw your symbolic approval on here

**Review by geralddejong — APPROVED** [review]

**Pull request merged**